### PR TITLE
Linting setup

### PR DESCRIPTION
**Pull request by sw00d**
# A brief description of the purpose of the changes contained in this PR.
Kind of a lot here. See the recipe `frontend-linting-setup.md` for setup.

- Recipe added
- precommit git hook added
- prettier + eslint set up

Then change some double quotes to singles and try to commit those single quotes (in a vue template or something). If the recipe is correct, you wont even be able to get to the committing part without it already being autocorrected to double quotes. If you do get to that point tho, our pre-commit hook should take care of it. 

There will almost certainly be bugs. Will try to run through this with Brian/Jess

### Handtesting for Jess/Brian/Harvey:
- [ ] do a git clone on this repo in a temporary dir on your computer
- [ ] cd into that repo and `git checkout linting-setup`
- [ ] run `make`
- [ ] Go the `recipes/frontend-linting-setup.md` and follow those directions for your IDE (pycharm or vscode)
- [ ] Then change some double quotes to singles and try to commit those single quotes (in a vue template or something). If the recipe is correct, you wont even be able to get to the committing part without it already being autocorrected to double quotes. If you do get to that point tho, our pre-commit hook should take care of it. 

# Checklist
- [x] Code review by me
- [ ] New code covered by automated tests
- [ ] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: recipes/frontend-linting-setup.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+
+Now a couple more files we need to make executable. 
+From root of project run:
+```
+ chmod +x ./bin/linting/format_frontend.sh  
+```
+```
+ chmod +x ./bin/linting/lint_frontend.sh  
+```
```
Comment (ckcollab) [inline recipes/frontend-linting-setup.md]: Hmm, this should be handled by `make` (aka `bin/fresh_install.sh`, specifically `bin/create_git_hooks.sh`) -- seems not necessary to include in the recipe, assuming the script handles things for you?

```sh
for f in ./bin/git_hooks/*
do
    mkdir -p .git/hooks/

    # get filename from full path
    filename=$(basename -- "$f")

    echo "making git hook -> ${f} into .git/hooks/${filename}"
    ln -sf "../../bin/git_hooks/${filename}" ".git/hooks/${filename}"
done
```

Comment (sw00d) [reply recipes/frontend-linting-setup.md]: Done

**Comment by sw00d** [comment]
Brian says: 

~~Seems to only work on vue templates/scripts/style but not in stylus style, .ts, .js, .py files~~ ✅ 


~~but the autosave is working and updating with prettier, but that works without the path being updated to ./src/frontend/.prettierrc~~ ✅ 

~~having to change the global prettier route feels a little bad (breaking prettier for other projects w/ different structure) and lead to a little confusion when I was fiddling with stuff (edited)~~ ✅ 

~~for instance if you opened a project in a sub folder (like opening frontend instead of the base route) prettier will throw errors /home/brian/work/skeletor/src/frontend/src/frontend/.prettierrc~~ ✅

**Comment by ckcollab** [comment]
ty guys! weeeeeee

**Pull request merged**